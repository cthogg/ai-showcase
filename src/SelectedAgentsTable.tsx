import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import { FEAgent } from "./types";
import { allCategories } from "./backendFrontendParser";
const useStyles = makeStyles({
  table: {
    minWidth: 650
  }
});

interface SelectedAgentsTableProps {
  agents: FEAgent[];
}

const categoryHeaders = allCategories.map(c => (
  <TableCell key={c} align="right">
    {c} Average
  </TableCell>
));
// FIXME: this needs to be repeated when another measure is added
const categoryAverage = (row: FEAgent, allCategories: string[]) =>
  allCategories.map(c => {
    // Find the category
    const statsForCategory = row.statistics.categories.find(
      rowForCategory => rowForCategory.category === c
    );
    // escapte the category
    return (
      <TableCell key={c} align="right">
        {statsForCategory?.statistics.averageMean}
      </TableCell>
    );
  });
// TODO: create row categories

const SelectedAgentsTable: React.FunctionComponent<SelectedAgentsTableProps> = ({
  agents
}: SelectedAgentsTableProps): JSX.Element => {
  const classes = useStyles();
  return (
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            {categoryHeaders}
          </TableRow>
        </TableHead>
        <TableBody>
          {agents.map(row => (
            <TableRow key={row.name}>
              <TableCell component="th" scope="row">
                {row.name}
              </TableCell>
              {categoryAverage(row, allCategories)}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default SelectedAgentsTable;
