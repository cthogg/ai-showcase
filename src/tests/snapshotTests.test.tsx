import React from "react";
import renderer from "react-test-renderer";
import ShowCaseGrid from "../ShowcaseGrid";
import { mockSelectedAgents } from "../mocks/mockSelectedAgents";
import SelectedAgentsTable from "../SelectedAgentsTable";

it("renders multiple questions correctly", () => {
  const tree = renderer
    .create(<ShowCaseGrid agents={mockSelectedAgents} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders agents table correctly", () => {
  const tree = renderer
    .create(<SelectedAgentsTable agents={mockSelectedAgents} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
