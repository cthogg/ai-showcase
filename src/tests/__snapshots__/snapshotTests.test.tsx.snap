// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders agents table correctly 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-189"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          memory
           Average
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          planning
           Average
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          logic
           Average
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          AlphaZero
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          33.0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          100.0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          86.0
        </td>
      </tr>
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          R2D3
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          79.0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          72.0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          32.7
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders multiple questions correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <ul
      className="MuiList-root makeStyles-root-104 MuiList-padding"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-131 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                aria-labelledby="checkbox-list-label-2"
                checked={false}
                className="PrivateSwitchBase-input-134"
                data-indeterminate={false}
                onChange={[Function]}
                tabIndex={-1}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          id="checkbox-list-label-2"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              AlphaZero
            </p>
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              <em>
                 
                Generalisation of AlphaGo Zero that can achieve, tabula rasa, superhuman performance in many challenging domains such as Chess, Shogi and Go.
                 
              </em>
            </p>
          </p>
        </div>
      </div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-131 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                aria-labelledby="checkbox-list-label-3"
                checked={false}
                className="PrivateSwitchBase-input-134"
                data-indeterminate={false}
                onChange={[Function]}
                tabIndex={-1}
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          id="checkbox-list-label-3"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              R2D3
            </p>
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              <em>
                 
                Making Efficient Use of Demonstrations to Solve Hard Exploration Problems.
                 
              </em>
            </p>
          </p>
        </div>
      </div>
    </ul>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  />
</div>
`;
